src/binary-reader.cc

```diff
@@ -1605,6 +1605,9 @@ static void read_custom_section(Context* ctx, uint32_t section_size) {
       uint32_t subsection_size;
       in_u32_leb128(ctx, &name_type, "name type");
       in_u32_leb128(ctx, &subsection_size, "subsection size");
+      size_t subsection_end = ctx->offset + subsection_size;
+      if (subsection_end > ctx->read_end)
+        RAISE_ERROR("invalid sub-section size: extends past end");
```

Comment:
But strange that BinaryEncoding.md has the new content but not the new section name.  Seems like that was a mistake.

If we really want to support old binaries with the previous name section we would need to build a parser that supports both, but we don't want to do that do we?

Do you happen to know what the current behavior is if this parser sees an old name section today?  I do seem to remember that it somehow skips over it.  Perhaps I could make that explicit in the code somehow.